lib/python/qmk/cli/new/keyboard.py

```diff
@@ -0,0 +1,139 @@
+        #   are reworked to change the letter case used internally,
+        #   lib/python/qmk/constants.py must be updated to match.
+        if mcu.lower() in SUPPORTED_MCUS:
+            mcu = mcu.lower()
+            arch = SUPPORTED_MCUS[mcu]
+        elif mcu.upper() in SUPPORTED_MCUS:
+            mcu = mcu.upper()
+            arch = SUPPORTED_MCUS[mcu]
+        else:
+            cli.log.error(mcu + " is not a valid microcontroller option.")
+            print("  Valid Options: ", ", ".join(SUPPORTED_MCUS))
+            exit(1)
+    else:
+        # Ask what microcontroller is being used
+        cli.echo("Select the microcontroller used:\n")
+        for i, mcu in enumerate(SUPPORTED_MCUS, 1):
+            cli.echo("    %s: %s (%s)", str(i).rjust(2, " "), mcu, SUPPORTED_MCUS[mcu])
+
+        mcu = question("Microcontroller: (1-%d): ", len(SUPPORTED_MCUS), answer_type=int)
+        # user-facing text is 1-indexed, but data is 0-indexed internally
+        mcu -= 1
+        # create a list of tuples from the dictionary
+        mcu_list = [(name, SUPPORTED_MCUS[name]) for name in SUPPORTED_MCUS]
+        arch = mcu_list[mcu][1]
+        mcu = mcu_list[mcu][0]
+    # Set the path to the MCU architecture's template files
+    template_arch_path = template_root_path / arch
+
+    if cli.config.new_keyboard.name:
+        user_name = cli.config.new_keyboard.name
+    else:
+        # Ask the user for their GitHub username
+        cli.echo("""What is your GitHub username?
+
+    This value will be pasted into the new files in copyright headers, and used
+    as the listed Keyboard Maintainer in the readme file.""")
+        user_name = question("GitHub Username: ")
+
+    # generate keyboard paths
+    keyboard_path = Path("keyboards") / keyboard
+
+    # check directories
+    if keyboard_path.exists():
+        cli.log.error('Keyboard %s already exists!', keyboard)
+        exit(1)
+    # if target directory is a subdirectory (keyboard project name contains a
+    #   forward slash), create the parent directory first
+    elif len(keyboard_path.parts) > 1:
+        keyboard_path.parent.mkdir(parents=True, exist_ok=True)
+    final_directory = keyboard_path.name
+
+    # create user directory with default keyboard files
+    shutil.copytree(template_base_path, keyboard_path, symlinks=True)
+    kb_c = keyboard_path / "keyboard.c"
+    kb_h = keyboard_path / "keyboard.h"
+    kb_c.rename(keyboard_path / Path(final_directory + ".c"))
+    kb_h.rename(keyboard_path / Path(final_directory + ".h"))
+
+    # copy architecture files
+    shutil.copy(template_arch_path / "config.h", keyboard_path)
+    shutil.copy(template_arch_path / "readme.md", keyboard_path)
+    shutil.copy(template_arch_path / "rules.mk", keyboard_path)
+    if arch == "ps2avrgb":
+        # only ps2avrgb keyboards require this file
+        shutil.copy(template_arch_path / "usbconfig.h", keyboard_path)
```

Comment:
The `quantum/template/ps2avrgb/usbconfig.h` is missing.

Reply:
Ah, removed by @fauxpark in #8584. Apparently it's not required anymore.